src/mono/wasm/Wasm.Build.Tests/TestMainJsProjectProvider.cs

```diff
@@ -14,156 +14,103 @@ public class TestMainJsProjectProvider : ProjectProviderBase
+        AssertBasicBundle(assertOptions);
 
-        TestUtils.AssertFilesExist(options.BundleDir, new[] { "run-v8.sh" }, expectToExist: options.HasV8Script);
-        AssertIcuAssets();
+        TestUtils.AssertFilesExist(assertOptions.BundleDir, new[] { assertOptions.MainJS });
+        if (assertOptions.IsBrowserProject)
+            TestUtils.AssertFilesExist(assertOptions.BundleDir, new[] { "index.html" });
+        TestUtils.AssertFilesExist(assertOptions.BundleDir, new[] { "run-v8.sh" }, expectToExist: assertOptions.HasV8Script);
 
-        string managedDir = Path.Combine(options.BundleDir, "_framework");
         string bundledMainAppAssembly =
-            options.UseWebcil ? $"{options.ProjectName}{WebcilInWasmExtension}" : $"{options.ProjectName}.dll";
-        TestUtils.AssertFilesExist(managedDir, new[] { bundledMainAppAssembly });
-
-        bool is_debug = options.Config == "Debug";
-        if (is_debug)
-        {
-            // Use cecil to check embedded pdb?
-            // AssertFilesExist(managedDir, new[] { $"{projectName}.pdb" });
-
-            //FIXME: um.. what about these? embedded? why is linker omitting them?
-            //foreach (string file in Directory.EnumerateFiles(managedDir, "*.dll"))
-            //{
-            //string pdb = Path.ChangeExtension(file, ".pdb");
-            //Assert.True(File.Exists(pdb), $"Could not find {pdb} for {file}");
-            //}
-        }
+            BuildTestBase.UseWebcil ? $"{assertOptions.ProjectName}{WebcilInWasmExtension}" : $"{assertOptions.ProjectName}.dll";
+        TestUtils.AssertFilesExist(assertOptions.BinFrameworkDir, new[] { bundledMainAppAssembly });
+    }
 
-        void AssertIcuAssets()
-        {
-            bool expectEFIGS = false;
-            bool expectCJK = false;
-            bool expectNOCJK = false;
-            bool expectFULL = false;
-            bool expectHYBRID = false;
-            switch (options.GlobalizationMode)
-            {
-                case GlobalizationMode.Invariant:
-                    break;
-                case GlobalizationMode.FullIcu:
-                    expectFULL = true;
-                    break;
-                case GlobalizationMode.Hybrid:
-                    expectHYBRID = true;
-                    break;
-                case GlobalizationMode.PredefinedIcu:
-                    if (string.IsNullOrEmpty(options.PredefinedIcudt))
-                        throw new ArgumentException("WasmBuildTest is invalid, value for predefinedIcudt is required when GlobalizationMode=PredefinedIcu.");
-                    TestUtils.AssertFilesExist(options.BundleDir, new[] { Path.Combine("_framework", options.PredefinedIcudt) }, expectToExist: true);
-                    // predefined ICU name can be identical with the icu files from runtime pack
-                    switch (options.PredefinedIcudt)
-                    {
-                        case "icudt.dat":
```

Comment:
If we are adding a list side by side with bools in `ProjectProviderBase.cs` then let's do it shorter:
```
         case GlobalizationMode.PredefinedIcu:
                if (string.IsNullOrEmpty(assertOptions.PredefinedIcudt))
                    throw new ArgumentException("WasmBuildTest is invalid, value for predefinedIcudt is required when GlobalizationMode=PredefinedIcu.");
                expected.Add(assertOptions.PredefinedIcudt);
                break;
```

otherwise we will add the file twice: once in `expected.Add(assertOptions.PredefinedIcudt);` and then as a result of evaluating the switch-case.
Sorry, I noticed it only in this PR.
I wonder why we are allowed to add a duplicate to the list and still the tests passes in the previous PR. `AssertFilesOnDisk` compares the counts of expected and actual so it should have failed. It seems we don't have a test with custom file that is from the runtime pack and that's how we overlooked it.

Reply:
Based on your feedback, I have removed the bools.